### DISCUSSION TITLE
Is there a way to share a database connection during getstaticprops?

### DISCUSSION THREAD

**Discussion by FBosler** [vercel/next.js#14386] [Help]
I am currently trying to use `getstaticprops` to enrich my dynamic pages with some SEO information at build time.

The CMS is only available via ssh-tunneling into a server and then extracting the data. Now, and I might be going wrong here, every single pageId creates their own tunnel, and pretty quickly they collide.

Is there a standard way to share a tunnel/database connection among all `getstaticprops` calls during build of pages with dynamic paths?

**Comment by desaijay315** [comment]
you can create the custom-server and then get all the logic of connections/api call and then call those API to the getStaticProps.

1. create customer server
2. Create connections - ssh /db connection
3. Create routes
4. create actions/api.js folder/file - make call to the routes you have created
5. Call those on particular pages

**Reply by FBosler** [reply]
Thank you!

It seems like a lot of duplication though (considering that I already have an API server). I guess I will just set up a couple of password protected routes and call those.

Hey btw. do I have to keep my credentials in the getStaticProps or can I have them somewhere else, import them and still have them removed after building?

Cheers

**Reply by desaijay315** [reply]
I would suggest to to perform above operations on your API server itself then. 

Set the credentials in your env variables inside **next.config.js**
